cloudevents/tests/test_with_sanic.py

```diff
@@ -0,0 +1,77 @@
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from cloudevents.sdk import marshaller
+from cloudevents.sdk import converters
+from cloudevents.sdk.event import v02
+
+from sanic import Sanic
+from sanic import response
+
+from cloudevents.tests import data as test_data
+
+
+m = marshaller.NewDefaultHTTPMarshaller()
+app = Sanic(__name__)
+
+
+@app.route("/is-ok", ["POST"])
+async def is_ok(request):
+    m.FromRequest(
+        v02.Event(),
+        dict(request.headers),
+        request.body,
+        lambda x: x
+    )
+    return response.text("OK")
+
+
+@app.route("/echo", ["POST"])
+async def echo(request):
+    event = m.FromRequest(
+        v02.Event(),
+        dict(request.headers),
+        request.body,
+        lambda x: x
+    )
+    hs, body = m.ToRequest(event, converters.TypeBinary, lambda x: x)
+    return response.text(body, headers=hs)
+
+
+def test_reusable_marshaller():
```

Comment:
This is great, I suspect that just calling `.FromRequest` 5 times would have caught my bug.

Reply:
well, maybe, but 10 is 2 times greater than 5, so we have double-insurance here =)